src/libsyntax/ast.rs

```diff
@@ -2524,7 +2524,7 @@ pub enum ItemKind {
     /// A type alias (`type`).
     ///
     /// E.g., `type Foo = Bar<u8>;`.
-    TyAlias(P<Ty>, Generics),
+    TyAlias(Generics, GenericBounds, Option<P<Ty>>),
```

Comment:
Looks like `type A;` is now semantically accepted as a free item.

Reply:
Indeed, but it should be due to foreign types, https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=47b47070b72642e26be4fe693525de8b

Reply:
Oh wait... semantically!... Will fix.

Reply:
Fixed.